core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -197,7 +197,7 @@ private static void setupLogging(Settings settings, Environment environment) {
 
     private static Tuple<Settings, Environment> initialSettings(boolean foreground) {
         Terminal terminal = foreground ? Terminal.DEFAULT : null;
-        return InternalSettingsPreparer.prepareSettings(EMPTY_SETTINGS, true, terminal);
+        return InternalSettingsPreparer.prepareSettingsAndEnvironment(EMPTY_SETTINGS, terminal);
```

Comment:
but i agree, this is already breaking. i will move it and simplify as you suggested